src/main/java/org/truffleruby/language/globals/UpdateVerbosityNode.java

```diff
@@ -9,31 +9,31 @@
  */
 package org.truffleruby.language.globals;
 
-import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
 import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.truffleruby.core.cast.BooleanCastNode;
+import org.truffleruby.core.cast.BooleanCastNodeGen;
 import org.truffleruby.language.RubyNode;
 
 public class UpdateVerbosityNode extends RubyNode {
 
+    @Child private BooleanCastNode booleanCastNode = BooleanCastNodeGen.create(null);
     @Child private RubyNode child;
 
+    private final ConditionProfile nilProfile = ConditionProfile.createBinaryProfile();
+
     public UpdateVerbosityNode(RubyNode child) {
         this.child = child;
     }
 
     @Override
     public Object execute(VirtualFrame frame) {
         final Object value = child.execute(frame);
-        validateVerbose(value);
-        return value;
-    }
 
-    @TruffleBoundary
-    private void validateVerbose(Object value) {
-        if (value instanceof Boolean || value == nil()) {
-            // OK
+        if (nilProfile.profile(value == nil())) {
+            return value;
         } else {
-            throw new UnsupportedOperationException();
+            return booleanCastNode.executeToBoolean(value);
```

Comment:
I was wondering if MRI casts as well and it seems it does:
```ruby
[1] pry(main)> $VERBOSE
=> false
[2] pry(main)> $VERBOSE = 42
=> 42
[3] pry(main)> $VERBOSE 
=> true
```

Could you maybe add a spec for this?

Reply:
I was a bit surprised to see there really wasn't anything testing `$VERBOSE`. Where would you want it? predefined_spec.rb?

Reply:
Yes.

Reply:
Spec added. I'll let you take a look at that before I merge.

Reply:
Thanks!